airflow/api/common/experimental/mark_tasks.py

```diff
@@ -48,11 +48,11 @@ def _create_dagruns(dag, execution_dates, state, run_type):
 
     for date in dates_to_create:
         dag_run = dag.create_dagrun(
-            run_id=f"{run_type}__{date.isoformat()}",
```

Comment:
What happend to run_id? I think we still need this, even if we don't use the prefix for detecting the type

Reply:
Oh I see, defaulted/handled in create_dagrun, nice.

Reply:
@ashb you can provide `run_id` or `run_type + execution_date`.  If you don't provide `run_id` it will be autogenerated. In most places, the `run_id` was created in the following way: `{run_type}__{execution_date.isoformat()}`

Reply:
Now I'm wondering, should we set the run_type using run_id? In example by checking the prefix. WDYT?

Reply:
If run_id is empty, it will be generated based on execution_date + run_type.

Reply:
Obtaining run_type from run_id may help because then we will always have run_type thus it can be used more effectively in index

Reply:
I think run_type should be non-null in the db, yes. So perhaps make run_type required, but run_id can be overridden.

We could detect run_type based on run_id, it would make it easier for anyone to upgrade, sure.